### PR TITLE
[sui-verifier] Small cleanup + comments

### PR DESCRIPTION
**Pull request by tnowacki**
- Removed `id_immutable` verifier pass, it is covered by `VersionedID` not having `drop`
- Removed check for types with `key` not having `drop`, it is covered by `VersionedID`
- Added doc comment to `struct_with_key_verifier`

### CODE REVIEW

**Comment by sblackshear** [comment]
These all make sense to me at a conceptual level--nice work! 

The only thing I would mention is that if we ever figure out how to get rid of the need to explicitly delete object ID's (which is kind of an ergonomics papercut at this point), we'll need this again. But I don't think we will be able to do that without big VM changes that aren't forthcoming (and maybe not even then).

**Comment by tnowacki** [comment]
> The only thing I would mention is that if we ever figure out how to get rid of the need to explicitly delete object ID's (which is kind of an ergonomics papercut at this point)

Ah, so you want `VersionedID` to have drop? I think its reasonable not to as it is the sort of normal way of preventing a reference from being written to (at least to me, but I probably don't have a normal, end-user Move perspective). 

What VM changes are you alluding to?

**Comment by sblackshear** [comment]
Purely hypothetical/magical VM changes, I don't have any concrete ideas in mind (except wanting to eliminate explicit ID deletion from the programmer's perspective)

File: crates/sui-verifier/src/struct_with_key_verifier.rs [hunk 1]
```diff
@@ -1,13 +1,23 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+//! This pass verifies necessary properties for Move Objects, i.e. structs with the `key` ability.
+//! The proprieties checked are
```
Comment (sblackshear) [inline crates/sui-verifier/src/struct_with_key_verifier.rs:5]:
```suggestion
//! The properties checked are
```

**Review by sblackshear — APPROVED** [review]

**Pull request merged**